compute/cloud-client/src/main/java/compute/CreateInstancesAdvanced.java

```diff
@@ -0,0 +1,405 @@
+
+      if (response.hasError()) {
+        System.out.println("Instance creation failed ! ! " + response);
+        return null;
+      }
+      System.out.println("Operation Status: " + response.getStatus());
+
+      return instancesClient.get(project, zone, instanceName);
+    }
+  }
+  // [END compute_instances_create_from_image]
+  // [END compute_instances_create_from_custom_image]
+  // [END compute_instances_create_from_image_plus_empty_disk]
+  // [END compute_instances_create_from_snapshot]
+  // [END compute_instances_create_from_image_plus_snapshot_disk]
+  // [END compute_instances_create_with_subnet]
+
+  // [START compute_instances_create_from_image]
+
+  /**
+   * Create a new VM instance with Debian 10 operating system.
+   *
+   * @param project      project ID or project number of the Cloud project you want to use.
+   * @param zone         name of the zone to create the instance in. For example: "us-west3-b"
+   * @param instanceName name of the new virtual machine (VM) instance.
+   * @return Instance object.
+   */
+  public static Instance createFromPublicImage(String project, String zone, String instanceName)
+      throws IOException, InterruptedException, ExecutionException {
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+      // List of public operating system (OS) images: https://cloud.google.com/compute/docs/images/os-details
+      Image image = imagesClient.getFromFamily("debian-cloud", "debian-10");
+      String diskType = String.format("zones/%s/diskTypes/pd-standard", zone);
+      Vector<AttachedDisk> disks = new Vector<>();
+      disks.add(diskFromImage(diskType, 10, true, image.getSelfLink()));
+      return createWithDisks(project, zone, instanceName, disks, "n1-standard-1",
+          "global/networks/default", null);
+    }
+  }
+  // [END compute_instances_create_from_image]
+
+  // [START compute_instances_create_from_custom_image]
+
+  /**
+   * Create a new VM instance with custom image used as its boot disk.
+   *
+   * @param project      project ID or project number of the Cloud project you want to use.
+   * @param zone         name of the zone to create the instance in. For example: "us-west3-b"
+   * @param instanceName name of the new virtual machine (VM) instance.
+   * @param customImage  link to the custom image you want to use in the form of:
+   *                     "projects/{project_name}/global/images/{image_name}"
+   * @return Instance object.
+   */
+  public static Instance createFromCustomImage(String project, String zone, String instanceName,
+      String customImage)
+      throws IOException, InterruptedException, ExecutionException {
+    String diskType = String.format("zones/%s/diskTypes/pd-standard", zone);
+    Vector<AttachedDisk> disks = new Vector<>();
+    disks.add(diskFromImage(diskType, 10, true, customImage));
+    return createWithDisks(project, zone, instanceName, disks, "n1-standard-1",
+        "global/networks/default", null);
```

Comment:
Why are these constants?  And if you are going to use a constant these days, why not use either E2 or N2 instances?

Reply:
Those are constant to make the example simpler. This is supposed to focus on using a custom disk image, so the machine type is just a constant, because configuring machine type is not demonstrated here. Reducing number of parameters passed to the function makes it easier to follow the important `customImage` parameter.

Switching to E2 in next commit.